link.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from scrapy.linkextractors import LinkExtractor
+from scrapy.http import HtmlResponse, TextResponse, Response
+from six.moves.urllib.parse import urlparse
+import glob
+from timeit import default_timer as timer
+import click
+import io
+
+
+def main():
+    start = timer()
+
+    url = 'http://scrapinghub.com/'
+    link_extractor = LinkExtractor()
+    total = 0
+    for files in glob.glob('sites/index*'):
+
+        f = (io.open(files, "r", encoding="ISO-8859-1"))
```

Comment:
@Parth-Vader I'm pretty sure that such method results in broken pages, here is why I think so.

wget just downloads the pages and saving them as bytes, and they all have different encodings. Most pages are in utf-8, but some are in latin-1 or different other encodings. latin-1 just happens to not throw errors when decoding many valid utf-8 pages, but the data is corrupted. Just to confirm, you can check ``sites/index.html.74`` in previous and in new version of the sites archive. Previous version is:
```
<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>天涯社区_全球华人网上家园</title>
...
```
and new one is:
```
<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>å¤©æ¶¯ç¤¾å<8c>º_å<85>¨ç<90><83>å<8d><8e>äººç½<91>ä¸<8a>å®¶å<9b>­</title>
...
```
as you can see, originally it was in utf-8 (as most pages), but after reading in latin-1 and encoding in utf-8 it got corrupted.

I think the best way to solve it is during page download phase, as you have the headers that can help to determine correct encoding. Scrapy already handles this correctly.